apps/builder/app/builder/features/style-panel/sections/box-shadows/box-shadow-content.tsx

```diff
@@ -49,51 +126,206 @@ export const BoxShadowContent = (props: BoxShadowContentProps) => {
       return;
     }
 
-    props.onEditLayer(props.index, layers);
+    onEditLayer(index, layers);
+  };
+
+  const handlePropertyChange = (
+    params: Partial<Record<keyof ExtractedBoxShadowProperties, StyleValue>>
+  ) => {
+    const newLayer = convertValuesToTupple(Object.assign(layerValues, params));
+    setIntermediateValue({
+      type: "intermediate",
+      value: toValue(newLayer),
+    });
+    onEditLayer(index, { type: "layers", value: [newLayer] });
+  };
+
+  const colorControlCallback: SetProperty = () => {
+    return (value) => {
+      handlePropertyChange({ color: value });
+    };
   };
 
   return (
-    <Flex
-      direction="column"
-      css={{
-        px: theme.spacing[5],
-        py: theme.spacing[5],
-        gap: theme.spacing[3],
-        minWidth: theme.spacing[30],
-      }}
-    >
-      <Label>
-        <Flex align={"center"} gap={1}>
-          Code
-          <Tooltip
-            variant="wrapped"
-            content={
-              <Text>
-                Paste a box-shadow value, for example:
-                <br />
-                <br />
-                0px 2px 5px 0px rgba(0, 0, 0, 0.2)
-              </Text>
+    <Flex direction="column">
+      <Grid
+        columns={2}
+        gap="2"
+        css={{
+          paddingTop: theme.spacing[5],
+          px: theme.spacing[9],
+        }}
+      >
+        <Flex direction="column">
+          <Label css={{ display: "inline" }}>X</Label>
+          <CssValueInputContainer
+            key="boxShadowOffsetX"
+            property="outlineOffset"
```

Comment:
Probably comment that outlineOffset is a "fake" property would be good to write in comment here pointing to the very good comment above